### PR TITLE
M3-387 hide edit and revoke functions if APIToken is an App Token

### PR DESCRIPTION
**Pull request by martinmckenna**
## Purpose

Editing and revoking individual tokens don't exist in the API, so for now, we can't give the user the ability to edit and revoke API tokens

### CODE REVIEW

**Comment by rmcintosh** [comment]
can one not expire an OAuth token via a `DELETE` to `/profile/tokens/$id`?

**Comment by rmcintosh** [comment]
I see the description in the item now... IMO it's weird for you guys to code around this, as it seems like buggy behavior to me.

**Comment by martinmckenna** [comment]
I think the plan is to put a ticket for the API to include that functionality and revisit this later

**Comment by Dorthu** [comment]
You're looking for [/profile/apps/:id DELETE](https://developers.linode.com/v4/reference/endpoints/profile/apps/$id)

**Comment by martinmckenna** [comment]
what's the difference between

/profile/apps/:id and /profile/tokens/:id @rmcintosh @Dorthu

**Comment by martinmckenna** [comment]
ok so per our discussion, the API docs are wrong and you can in fact delete an individual App Token with `profile/apps/:token`

will implement a fix

**Comment by rjstires** [comment]
@rmcintosh FWFW there was a miscommunication via Redoc that resulted in this PRs state. 
image
PATS (/tokens) and APPS (/apps) can be revoked. PATS can be edited, APPS cannot. 
@martinmckenna is updating this PR to reflect that.

**Comment by martinmckenna** [comment]
@rjstires @rmcintosh @Dorthu App tokens can now be revoked in 906409b

File: src/features/profile/APITokens/APITokens.tsx [hunk 1]
```diff
@@ -303,13 +306,20 @@ export class APITokens extends React.Component<CombinedProps, State> {
...
 
-  revokeToken = () => {
+  revokePersonalAccessToken = () => {
     const { dialog } = this.state;
     Axios.delete(`${API_ROOT}/profile/tokens/${dialog.id}`)
       .then(() => { this.closeRevokeDialog(); })
       .then(() => this.requestTokens());
   }
 
+  revokeAppToken = () => {
```
Comment (rjstires) [inline src/features/profile/APITokens/APITokens.tsx]: Let's move this into `services/linodes` and use the new Request generator. Let me know if you need a brief overview of how it works.

**Comment by martinmckenna** [comment]
@rjstires requested changes have been made

**Pull request merged**